app/src/main/java/com/kickstarter/viewmodels/ProjectViewModel.kt

```diff
@@ -589,6 +589,46 @@ interface ProjectViewModel {
                     .map { p -> if (p.isStarred) R.drawable.icon__heart else R.drawable.icon__heart_outline }
                     .subscribe(this.heartDrawableId)
 
+            //Tracking
+            this.rewardsButtonText
```

Comment:
It seems this file has gotten pretty large. I realize this may be stylistic, but in an effort to have this class take on a bit less responsibility, would it make sense (or even be possible?? I don't know the android things) to extract a class that's responsible for doing tracking in this view? I.e. its purpose is to subscribe to all relevant  emissions (from `nativeCheckoutProject` and whatever else we'd want to listen to) and track the correct things? Might just make it one less responsibility to test/worry about in the `ProjectViewModel`

Reply:
Hmm...I like the idea. So it'd be basically an extension of the ProjectViewModel, right? But it needs to have access to the inputs and outputs and whatever helper vars I create 🤔 

But yeah this screen and the pledge screen basically hold all the majority of the business logic of the app.

Reply:
wow just tried to pseudo code what I think the class would look like but was failing, hard lol

I didn't realize that you'd have to also pass in the PublishSubjects `viewRewardsClicked`, `cancelPledgeClicked`, `contactCreatorClicked`, `updatePledgeClicked` and `nativeProjectActionButtonClicked` in addition to all the observables so maybe this pattern doesn't make a ton of sense :X Sorry!

Reply:
No need to be sorry! I would like move all the pledge sheet logic to their own ViewHolder (and corresponding ViewModel) but haven't figured that out yet 🤔 🔜